### PR TITLE
Use separate jenkins build and test process

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
jenkins, test this with java 11 please

**Comment by yuzhu** [comment]
jenkins, test this with java 11 please

**Comment by yuzhu** [comment]
jenkins, test this with java 11 please

**Comment by yuzhu** [comment]
@gpang FYI, compile with java 8, test with java 11.

File: dev/jenkins/build.sh [hunk 1]
```diff
@@ -68,8 +68,14 @@ if [ "$RUN_MAVEN" == "true" ]; then
   then
     ALLUXIO_BUILD_FORKCOUNT=4
   fi
-
-  mvn -Duser.home=/home/jenkins -T 4C clean install -Pdeveloper -Dmaven.javadoc.skip -Dsurefire.forkCount=${ALLUXIO_BUILD_FORKCOUNT} ${mvn_args} $@
+  JAVA_HOME_BACKUP=${JAVA_HOME}
```
Comment (gpang) [inline dev/jenkins/build.sh:71]: Can you add a comment that mentions that these default env variables are pointing to java 11 initially?

Comment (yuzhu) [reply dev/jenkins/build.sh:71]: they might. This file is shared between java 8 and java 11 tests. in both cases, we use java 8 to compile and use the image default (java8 or java 11 ) to run the test

Comment (yuzhu) [reply dev/jenkins/build.sh:71]: i will add a comment anyways

**Comment by yuzhu** [comment]
@gpang  PTAL

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**